phac-nml/irida-next
# Groups dashboard searching sorting

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Added search by group name or puid through text input field above tree component.  When search is active, the tree becomes "flat".

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/user-attachments/assets/0755ab0a-f98d-4688-b02f-181e3a60548d)

![image](https://github.com/user-attachments/assets/778481fa-6dff-4de6-9347-f3dc03b8b985)

![image](https://github.com/user-attachments/assets/9a3904b1-4062-445f-bc4b-7894b7f4b7ef)

**Empty search result**
![image](https://github.com/user-attachments/assets/8788b6b1-e09b-432d-b3b0-d30c383223ec)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Go to groups dashboard page
2. Expand `Streptococcus` row and should have child groups
3. Filter by `strep` and should see a flatten listing of all stretococcus groups
4. Try also filter by PUID

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
